cashctrl_ledger/extended_ledger.py

```diff
@@ -0,0 +1,275 @@
+"""This module contains "hackish" code that violates the limitations of the CashCtrl."""
+
+from typing import Union
+import numpy as np
+import pandas as pd
+from .ledger import CashCtrlLedger
+
+
+class CashCtrlLedgerExtended(CashCtrlLedger):
+    """Class that inherits the CashCtrlLedger class and violates
+    the limitations of the CashCtrl.
+
+    See README on https://github.com/macxred/cashctrl_ledger for overview and
+    usage examples.
+    """
+
+    _transitory_account = None
+
+    # ----------------------------------------------------------------------
+    # Constructor
+
+    def __init__(self, transitory_account: int):
+        super().__init__()
+        self.transitory_account = transitory_account
+
+    # ----------------------------------------------------------------------
+    # Accounts
+
+    @property
+    def transitory_account(self) -> int:
+        """Transitory account for balancing entries.
+
+        Some complex transactions can not be mapped to CashCtrl. We split such transactions
+        into multiple simpler transactions. The balance of each simple transaction is booked
+        onto the transitory account, where the combination of all postings originating from
+        the same complex transactions should sum up to zero.
+
+        Raises:
+            ValueError: If transitory_account is not set or the account does not exist.
+
+        Returns:
+            int: The transitory account number.
+        """
+        if self._transitory_account is None:
+            raise ValueError("transitory_account is not set.")
+        if self._transitory_account not in set(
+            self._client.list_accounts()["number"]
+        ):
+            raise ValueError(
+                f"The transitory account {self._transitory_account} does not exist."
+            )
+        account_currency = self._client.account_to_currency(self._transitory_account)
+        if account_currency != self.base_currency:
+            raise ValueError(
+                f"The transitory account {self._transitory_account} must be "
+                f"denominated in {self.base_currency} base currency, not "
+                f"{account_currency}."
+            )
+        return self._transitory_account
+
+    @transitory_account.setter
+    def transitory_account(self, value: int):
+        self._transitory_account = value
+
+    # ----------------------------------------------------------------------
+    # Ledger
+
+    def sanitize_ledger(self, ledger: pd.DataFrame) -> pd.DataFrame:
+        """Sanitizes the ledger DataFrame by splitting multi-currency transactions
+        and ensuring FX adjustments.
```

Comment:
Description is not clear. What exactly does the method do? Maybe worthwhile to mention that we are overriding a generic method to comply with specific limitations of CashCtrl.